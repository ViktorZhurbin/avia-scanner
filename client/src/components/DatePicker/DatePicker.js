import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames/bind';

import Loading from '../Loading';
import Dropdown from '../Dropdown';
import { classNamesPropType } from '../../entities/propTypes';

import styles from './DatePicker.css';

const cx = cl.bind(styles);

const CalendarPromise = import(/* webpackChunkName: "calendar" */ 'react-calendar');
const Calendar = React.lazy(() => CalendarPromise);

const DatePicker = ({
    trigger,
    onSelect,
    classNames,
}) => {
    const [date, setDate] = useState(new Date());

    const handleDateSelect = (newDate) => {
        setDate(newDate);
        onSelect(newDate);
    };

    return (
        <Dropdown
            classNames={{ ...classNames }}
            trigger={trigger}
        >
            <React.Suspense fallback={<Loading />}>
                <div className={cx('calendarModal')}>
                    <Calendar
                        value={date}
                        onChange={handleDateSelect}
                    />
                </div>
                <div className={cx('modalOverlay')} />
            </React.Suspense>
        </Dropdown>

    );
};

DatePicker.propTypes = {
    trigger: PropTypes.node.isRequired,
    onSelect: PropTypes.func.isRequired,
    classNames: classNamesPropType,
};

DatePicker.defaultProps = {
    classNames: null,
};

export default DatePicker;
