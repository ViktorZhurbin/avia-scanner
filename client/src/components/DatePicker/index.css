.calendarModal {
    display: block;
    z-index: 100;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    max-width: 100%;
    max-height: 100%;
    box-shadow: 0 0 2px 1px rgba(0, 0, 0, 0.15);
}

.modalOverlay {
    z-index: 50;
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.3);
}
