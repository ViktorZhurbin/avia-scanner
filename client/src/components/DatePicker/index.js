import React from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames/bind';

import Loader from '../Loader';
import Dropdown from '../Dropdown';
import { classNamesPropType } from '../../entities/propTypes';

import styles from './index.css';

const cx = cl.bind(styles);

const CalendarPromise = import(/* webpackChunkName: "calendar" */ 'react-calendar');
const Calendar = React.lazy(() => CalendarPromise);

class DatePicker extends React.PureComponent {
    static propTypes = {
        trigger: PropTypes.node.isRequired,
        onSelect: PropTypes.func.isRequired,
        classNames: classNamesPropType,
    }

    static defaultProps = {
        classNames: null,
    }

    state = {
        date: new Date(),
    }

    handleDateSelect = (date) => {
        const { onSelect } = this.props;

        this.setState({
            date,
        }, () => onSelect(date));
    }

    render() {
        const {
            trigger,
            classNames,
        } = this.props;

        return (
            <Dropdown
                classNames={classNames}
                trigger={trigger}
            >
                <React.Suspense fallback={<Loader />}>
                    <div className={cx('calendarModal')}>
                        <Calendar
                            value={this.state.date}
                            onChange={this.handleDateSelect}
                        />
                    </div>
                    <div className={cx('modalOverlay')} />
                </React.Suspense>
            </Dropdown>

        );
    }
}

export default DatePicker;
