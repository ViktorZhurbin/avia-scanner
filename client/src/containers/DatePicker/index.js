import React from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames';
import ReactOutsideEvent from 'react-outside-event';

import Calendar from '../../components/Calendar';

import styles from './index.css';

const cx = cl.bind(styles);

class DatePicker extends React.Component {
    static propTypes = {
        children: PropTypes.node.isRequired,
    }

    state = {
        isOpen: false,
    }

    onToggle = () => {
        const { isOpen } = this.state;

        this.setState(() => ({
            isOpen: !isOpen,
        }));
    }

    handleKeyPress = (event) => {
        if (event.key === 'Enter') {
            this.onToggle();
        }
    }

    onOutsideEvent = () => {
        this.setState(() => ({
            isOpen: false,
        }));
    }

    render() {
        const { children } = this.props;

        return (
            <div className={cx('container')}>
                <div
                    className={cx('trigger')}
                    role="button"
                    data-toggle="dropdown"
                    tabIndex="0"
                    onClick={this.onToggle}
                    onKeyPress={this.handleKeyPress}
                >
                    {children}
                </div>
                <div className={cx('calendar')}>
                    <Calendar />
                </div>
            </div>
        );
    }
}

export default ReactOutsideEvent(DatePicker, ['click']);
