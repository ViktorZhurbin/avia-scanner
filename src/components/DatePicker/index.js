import React from 'react';
import PropTypes from 'prop-types';
import { SingleDatePicker } from 'react-dates';
import { ICON_AFTER_POSITION, OPEN_UP } from 'react-dates/constants';

class DatePicker extends React.Component {
    static propTypes = {
        handleChange: PropTypes.func.isRequired,
        date: PropTypes.object, // eslint-disable-line
        placeholder: PropTypes.string,
    }

    static defaultProps = {
        placeholder: null,
        date: null,
    };

    state = {
        focused: false,
    }

    onDateChange = (date) => {
        this.props.handleChange('departure', date);
    }

    render() {
        const { date, placeholder } = this.props;

        return (
            <SingleDatePicker
                date={date}
                id="date"
                onDateChange={this.onDateChange}
                focused={this.state.focused}
                onFocusChange={({ focused }) => this.setState({ focused })}
                required
                noBorder
                placeholder={placeholder}
                showClearDate={Boolean(date)}
                showDefaultInputIcon={!date}
                inputIconPosition={ICON_AFTER_POSITION}
                openDirection={OPEN_UP}
                withPortal
                hideKeyboardShortcutsPanel
                transitionDuration={100}
                numberOfMonths={1}
            />
        );
    }
}

export default DatePicker;
