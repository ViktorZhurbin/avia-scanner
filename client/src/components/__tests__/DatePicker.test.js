import React from 'react';
import { shallow } from 'enzyme';

import DatePicker from '../DatePicker';

const props = {
    onSelect: () => null,
};

describe('Render DatePicker', () => {
    it('render correctly date component', () => {
        const component = shallow(
            <DatePicker
                {...props}
                trigger={<div />}
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
