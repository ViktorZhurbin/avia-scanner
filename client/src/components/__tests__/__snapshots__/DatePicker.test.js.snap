// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DatePicker render correctly date component 1`] = `ShallowWrapper {}`;
